JFCaBa/shooting-game-cms
# Error handle and pagination

# Authentication and Pagination Improvements

## Changes Overview
1. Added proper authentication handling with React Context
2. Implemented server-side pagination for all resource lists
3. Added filtering capabilities to data tables
4. Improved error handling and loading states
5. Enhanced UI/UX for tables and navigation

## Authentication Changes
- Created new AuthContext for centralized auth state management
- Improved token handling and unauthorized access flows
- Added loading states to login form
- Better error messages for authentication failures
- Proper redirect handling for 401 errors

## Pagination Implementation
- Added server-side pagination to all list endpoints:
  - Players
  - Drones
  - Achievements
  - Rewards
  - Token Balances
- Implemented pagination metadata (page, limit, total, totalPages)
- Added pagination controls to DataTable component
- Default page size of 10 items with configurable limits

## Data Table Improvements
- Added new FilterDropdown component for column filtering
- Improved sorting functionality
- Added filter types: text, number, date, select
- Better table header layout with filter controls
- Enhanced mobile responsiveness

## API Changes
- Updated all list endpoints to support pagination params
- Added validation for pagination parameters
- Created utility functions for pagination handling
- Improved error handling in API responses
- Better type checking and validation

## Backend Updates
- Added pagination support to all controllers
- Implemented filter parameter handling
- Added proper error responses
- Created pagination utility functions
- Updated route validators

## UI Enhancements
- Improved header and navigation layout
- Better loading states
- Enhanced error messages
- More consistent styling
- Better mobile responsiveness

## How to Test
1. Check authentication flow:
   - Login/logout
   - Token expiration
   - Unauthorized access handling

2. Test pagination:
   - Navigate through pages
   - Change page size
   - Check total counts

3. Verify filtering:
   - Test different filter types
   - Combined filters
   - Clear filters

4. Verify data loading:
   - Loading states
   - Error handling
   - Empty states

## Note
This PR requires backend changes to be deployed first as it depends on the new pagination endpoints.
